cli-commands/etherlime-test/etherlime-coverage.js

```diff
@@ -0,0 +1,220 @@
+var Mocha = require("mocha");
+var chai = require("chai");
+var originalRequire = require("original-require");
+let timeTravel = require('./time-travel');
+let events = require('./events');
+
+let accounts = require('./../ganache/setup.json').accounts;
+let compiler = require('./../compiler/compiler');
+let ethers = require('ethers');
+let CustomReporter = require('./gas-logger/gas-reporter');
+
+chai.use(require("./assertions"));
+
+const ganache = require('../ganache/ganache');
+const child_process = require('child_process');
+const ProviderEngine = require("web3-provider-engine");
+const RpcProvider = require("web3-provider-engine/subproviders/rpc.js");
+const { CoverageSubprovider } = require("@0x/sol-coverage");
+const SolCompilerArtifactAdapter = require('@0x/sol-coverage').SolCompilerArtifactAdapter;
+
+const defaultFromAddress = "0xd9995bae12fee327256ffec1e3184d492bd94c31";
+const isVerbose = true;
+let artifacts = `./coverage-artifacts`;
+const Compiler = require('@0x/sol-compiler').Compiler;
+const provider = new ProviderEngine();
+const dir = require('node-dir');
+const path = require('path');
+const fs = require('fs')
+var istanbul = require('istanbul');
+const shell = require('shelljs');
+
+const runCoverage = async (files, port, runs, solcVersion, buildDirectory, workingDirectory, shouldOpenCoverage) => {
+	var mochaConfig = { 'useColors': true };
+	let mocha = createMocha(mochaConfig, files);
+	mocha.reporter(CustomReporter, { port });
+
+	files.forEach(function (file) {
+		delete originalRequire.cache[file];
+
+		mocha.addFile(file);
+	});
+
+	const coverageProvider = await prepareCoverage(workingDirectory, port)
+	await setJSTestGlobals(port, coverageProvider);
+
+	await compiler.run('.', undefined, solcVersion, false, undefined, false, true, buildDirectory, workingDirectory);
+	await compilationCoverageArtifacts(solcVersion, workingDirectory, runs, buildDirectory);
+	await runMocha(mocha);
+	await generateCoverageReports(shouldOpenCoverage);
+}
+
+// Compile contracts in desired format in order to pass them to coverage library
+const compilationCoverageArtifacts = async (solcVersion, workingDirectory, runs, buildDirectory) => {
+	const compilerOptions = {
+		contractsDir: `${process.cwd()}/${workingDirectory}`,
+		artifactsDir: `${process.cwd()}/${artifacts}`,
+		compilerSettings: {
+			outputSelection: {
+				['*']: {
+					['*']: ['abi', 'evm.bytecode.object', 'evm.deployedBytecode.object'],
+				},
+			},
+			optimizer: {
+				enabled: runs ? true : false,
+				runs: runs
+			}
+		},
+		contracts: '*',
+		solcVersion: solcVersion,
+	};
+
+	const compiler = new Compiler(compilerOptions);
+
+	console.log('Preparing coverage environment and building artifacts...');
+
+	await compiler.compileAsync();
+
+
+	await prepareCoverageBuildedFiles(buildDirectory)
+
+}
+
+const prepareCoverageBuildedFiles = async (buildDirectory) => {
```

Comment:
There is no word `builded` - `built` is the correct one

Reply:
Also for me the name of this method should be somewhat suggesting that there is an overwrite of part of the files. Like - `overrideCoverageBytecodes` or something like that